diff --git a/src/main/java/org/neo4j/examples/social/TheMatrix.java b/src/main/java/org/neo4j/examples/social/TheMatrix.java
index 3a0faea..fb86f9f 100644
--- a/src/main/java/org/neo4j/examples/social/TheMatrix.java
+++ b/src/main/java/org/neo4j/examples/social/TheMatrix.java
@@ -255,9 +255,9 @@ public class TheMatrix
      */
     public Node createPersonNode( String name )
     {
-        // TODO: update in Step 3
         Node person = graphDb.createNode();
         person.setProperty( PERSON_NAME, name );
+        graphDb.index().forNodes( "Persons" ).add( person, PERSON_NAME, name );
         return person;
     }
 
@@ -300,9 +300,9 @@ public class TheMatrix
      */
     public Node createInterestNode( String interest )
     {
-        // TODO: update in Step 3
         Node interestNode = graphDb.createNode();
         interestNode.setProperty( "name", interest );
+        graphDb.index().forNodes( "Interests" ).add( interestNode, "interest", interest );
         return interestNode;
     }
 
@@ -331,7 +331,6 @@ public class TheMatrix
      */
     public Node lookupPerson( String name )
     {
-        // TODO: implement this in Step 3
-        return null;
+        return graphDb.index().forNodes( "Persons" ).get( PERSON_NAME, name ).getSingle();
     }
 }
